background playback capable

andrewzadrozny opened the pull request:
radio play can now run when app is in the background or phone is asleep. also implemented jake's suggested changes

andrewzadrozny committed:
updated playlist image feature

now can search for image urls from discogs, lastfm, and itunes

andrewzadrozny committed:
background playback capable

followed jake's suggestions and changed radio playback so that it can play in the background

jakebromberg commented on BasicMusicPlayer/app/src/main/java/com/example/basicmusicplayer/PlayerActivity.kt:
```diff
@@ -45,33 +51,25 @@ class PlayerActivity : AppCompatActivity() {
                 // updated playlist values
-                val updatedSubList = playlistManager.fetchLittlePlaylist()
+                var updatedSubList = playlistManager.fetchLittlePlaylist()
+                if (updatedSubList.size != 6){
```
Could you extract the 6 into a named constant?

jakebromberg replied:
Are you only showing 6 entries? Any reason not to show more?

andrewzadrozny replied:
the 6 entries is just for the updated segment of the playlist. the rest of the playlist is visible, just staying constant. 6 seemed like a good value in terms of the range of entries that could potentially be edited, but this value could be changed.

jakebromberg commented on BasicMusicPlayer/app/src/main/java/com/example/basicmusicplayer/PlayerActivity.kt:
```diff
@@ -118,14 +115,35 @@ class PlayerActivity : AppCompatActivity() {
+            showContentView()
+        }
+
+
```
Please be consistent with spacing, here and elsewhere.

jakebromberg commented on BasicMusicPlayer/app/src/main/java/com/example/basicmusicplayer/PlayerActivity.kt:
```diff
@@ -184,57 +202,27 @@ class PlayerActivity : AppCompatActivity() {
+            println("said audio isnt playing")
+            loadingIndicator.visibility = View.VISIBLE
+            startService(Intent(this, AudioPlaybackService::class.java))
+            loadingIndicator.visibility = View.GONE // Hide the loading indicator
```
You hide the loading indicator in both branches. Extract this line into the enclosing scope?

jakebromberg commented on BasicMusicPlayer/app/src/main/java/data/artwork/ImageLoader.kt:
```diff
@@ -1,14 +1,32 @@
+            it.setBounds(0, 0, it.intrinsicWidth, it.intrinsicHeight)
+        }
+
+        */
```
Remove this?

jakebromberg commented on BasicMusicPlayer/app/src/main/java/data/artwork/discogs/DiscogsArtistsResults.kt:
```diff
@@ -0,0 +1,25 @@
+
+data class DiscogsArtistsResults(val pagination: Pagination,
+                                 val results: List<Result>
+)
```
Might be more readable to reformat this like:
```
data class DiscogsArtistsResults(
    val pagination: Pagination,
    val results: List<Result>
)
```

jakebromberg commented on BasicMusicPlayer/app/src/main/java/data/artwork/itunes/ITunesResults.kt:
```diff
@@ -0,0 +1,35 @@
+
+data class ITunesResults(val results: List<ITunesItem>)
+
+data class ITunesItem(val wrapperType: String,
```
Reformat

andrewzadrozny committed:
updated frontend and audioplayback

design implemented, primary audio focus, cleaned up code

jakebromberg commented on BasicMusicPlayer/app/src/main/java/com/example/basicmusicplayer/PlayerActivity.kt:
```diff
@@ -160,81 +112,142 @@ class PlayerActivity : AppCompatActivity() {
+                    // now that we know there is an update, we fetch the new entries with their art
+                    val updatedSublistWithImages = playlistManager.fetchLittlePlaylistWithImages()
+                    println("update time")
+                    runOnUiThread {
```
Could you extract the body of this closure into a new method to keep this method shorter?

jakebromberg commented on BasicMusicPlayer/app/src/main/java/com/example/basicmusicplayer/PlayerActivity.kt:
```diff
@@ -160,81 +112,142 @@ class PlayerActivity : AppCompatActivity() {
+                    }
+                    // new entry
+                    else {
+                        newEntry = true
```
If `editPlaylist` and `newEntry` are mutually exclusive, would it make sense to just use one variable?

jakebromberg commented on BasicMusicPlayer/app/src/main/java/data/artwork/PlaylistImager.kt:
```diff
@@ -1,43 +1,223 @@
 
-                    val imageUrl = searchResult.results.firstOrNull()?.thumb
+    //fetches the image url from itunes
+    suspend fun fetchItunesImage(playcut: PlaylistDetails): String? {
```
Maybe name this `fetchItunesImageURL` since it's not actually fetching the image?

andrewzadrozny committed:
adjusted audio playback and error handling

playback is now loaded with app and play/pause acts as unmute/mute. if the stream is muted for 1 minute, the stream is released and will be restarted when user presses play again. also introduced handling for network loss
